### ISSUE TITLE
Firestore read/write optimization or best practices 

### ISSUE THREAD

**Issue by changer6360** [peterfriese/MakeItSo#6]
Hi Peter,

There is something that almost nobody is explaining/showing, at least with real code - how to optimize reads and writes to Firestore. 
Currently, you know, if using Firestore, you always download all of the data (of a collection) on app launch, no matter if you are using one time fetch or a listener. Potentially you wanna have a lot of users, which on their side can have a lot of tasks. So, when these users start to use your app, you are going to have thousands of calls per day even though they might not be needed (no change in the data whatsoever). 
My idea is if you can show us a valid way to cache the data and call for a refresh only when needed or there is a new data on the server.

**Comment by SebastianBO** [comment]
Great idea!

**Comment by peterfriese** [comment]
Thanks for the suggestion - I will see if I can pull together some details for an article.

**Comment by RWOverdijk** [comment]
Oh yes, I'd love to know this. I don't know if this is useful at all, but coming from web I know that firestore does already try to optimize some of this for you. For iOS I think it has support for offline enabled by default and you can define the source of your data (cache/server).

**Comment by changer6360** [comment]
From my experience even with _cache_ being set, the first call will fetch everything regardless the setting. Basically rebuilding/creating the cache on every app launch. And then you can filter the results with updated, modified, etc. 
The closest to my question that I've found was using modify timestamps for every record, keep everything locally and check regularly for changes, but I kind of lost interest because the idea of firebase is to be easy an out of the box.

**Comment by RWOverdijk** [comment]
That's true! I didn't mean the initial fetch, sorry. I meant every subsequent cache. Even pagination actively counts reads, so that'll be hard. But after the first time it could be optimised.

But firestore won't be a solution for everything, that's true.
